src/icalendar/tests/test_encoding.py

```diff
@@ -88,3 +96,48 @@ def test_unicode_parameter_name(self):
             + b'\xc3\xa4\xc3\xb6\xc3\xbc\xc3\x9f\xc3\x84\xc3\x96\xc3\x9c\r\n'
             + b'END:VEVENT\r\nEND:VCALENDAR\r\n'
         )
+
+@pytest.mark.parametrize('event_name', [
+    # Non-unicode characters in summary
+    'issue_64_event_with_non_unicode_summary',
+    # Unicode characters in summary
+    'issue_64_event_with_unicode_summary',
+    # chokes on umlauts in ORGANIZER
+    'issue_101_icalendar_chokes_on_umlauts_in_organizer'
+])
+def test_events_unicoded(events, event_name):
+    '''Issue #64 - Event.to_ical() fails for unicode strings
+       Issue #101 - icalendar is choking on umlauts in ORGANIZER
+
+    https://github.com/collective/icalendar/issues/64
+    https://github.com/collective/icalendar/issues/101
+    '''
+    event = getattr(events, event_name)
+    assert event.to_ical() == event.raw_ics
+
+@pytest.mark.parametrize('timezone', [
```

Comment:
I would like to change this and add a parametrized fixture that takes the time zone string as an argument, so we make sure we have one place in which we list all the different time zone implementations.

https://github.com/niccokunzmann/python-recurring-ical-events/blob/7f6aeed4729b97981c22ef0f481029cca225ae76/test/conftest.py#L120

In conftest: 

```
# for parametrizing fixtures, see https://docs.pytest.org/en/latest/fixture.html#parametrizing-fixtures
@pytest.fixture(params=[pytz.timezone, tz.gettz, zoneinfo.ZoneInfo], scope="module")
def TZ(request):
    return request.param()
```

Here:

```
def test_parses_event_with_non_ascii_tzid_issue_237(events, TZ):
...
    expected = datetime.datetime(2017, 5, 11, 13, 30, tzinfo=TZ('America/Sao_Paulo'))
```

1. This way, we can add a timezone implementation easily in one place for all tests.
2. We can use this fixture in the future, if we were to specify a timezone parameter for the parsing of the ics files.

What do you think?

Reply:
https://github.com/collective/icalendar/runs/8273351016?check_suite_focus=true#step:6:58 would be solved by passing a parameter to the parser of which implementation to use.

Reply:
Should that be in the commit with the test or a separate one?